packages/stitch/tests/mergeComputedFields.test.ts

```diff
@@ -154,3 +154,866 @@ describe('merge computed fields via config', () => {
     ]);
   });
 });
+
+describe('test merged composite computed fields', () => {
+  const schemaA = makeExecutableSchema({
+    typeDefs: /* GraphQL */ `
+      interface I {
+        id: ID!
+        value: Int!
+      }
+
+      type T implements I {
+        id: ID!
+        value: Int!
+      }
+
+      type U {
+        id: ID!
+        value: Int!
+      }
+
+      union W = T | U
+
+      type Query {
+        byId(id: ID!): T
+        uById(id: ID!): U
+      }
+    `,
+    resolvers: {
+      T: {
+        value: (obj: { id: string }) => parseInt(obj.id),
+      },
+      U: {
+        value: (obj: { id: string }) => parseInt(obj.id),
+      },
+      I: {
+        __resolveType: () => 'T',
+      },
+      Query: {
+        byId: (_: never, { id }: { id: string }) => ({ id }),
+        uById: (_: never, { id }: { id: string }) => ({ id }),
+      },
+    },
+  });
+
+  describe('object-valued computed field', () => {
+    const schemaB = makeExecutableSchema({
+      typeDefs: /* GraphQL */ `
+        interface I {
+          id: ID!
+          value2: Int!
+        }
+
+        type T implements I {
+          id: ID!
+          next: T!
+          value2: Int!
+        }
+
+        type U {
+          id: ID!
+          value2: Int!
+        }
+
+        type V {
+          id: ID!
+        }
+
+        union W = T | U
+
+        union X = T | V
+
+        input TInput {
+          id: ID!
+          value: Int
+        }
+
+        type Query {
+          byRepresentation(representation: TInput!): T
+          uByRepresentation(representation: TInput!): U
+        }
+      `,
+      resolvers: {
+        T: {
+          next: (obj: { id: string; value: number }) => ({ id: `${obj.value + 1}` }),
+        },
+        U: {
+          value2: (obj: { id: string }) => parseInt(obj.id),
+        },
+        Query: {
+          byRepresentation: (
+            _: never,
+            { representation: { id, value } }: { representation: { id: string; value: number } }
+          ) => ({ id, value }),
+          uByRepresentation: (
+            _: never,
+            { representation: { id, value } }: { representation: { id: string; value: number } }
+          ) => ({ id, value }),
+        },
+      },
+    });
+
+    const gatewaySchema = stitchSchemas({
+      subschemas: [
+        {
+          schema: schemaA,
+          merge: {
+            T: {
+              selectionSet: '{ id }',
+              fieldName: 'byId',
+              args: ({ id }) => ({ id }),
+            },
+            U: {
+              selectionSet: '{ id }',
+              fieldName: 'uById',
+              args: ({ id }) => ({ id }),
+            },
+          },
+        },
+        {
+          schema: schemaB,
+          merge: {
+            T: {
+              selectionSet: '{ id }',
+              fieldName: 'byRepresentation',
+              args: ({ id, value }) => ({ representation: { id, value } }),
```

Comment:
How do we expect `value` here?
Because it is needed for `next` only, no?

Reply:
Yeah, should have made clear by proper typing `value` is optional and will only be filled if available b/c a) `value` was explicitly queried, or b) the computed field was queried. Fixed the typing and added a test to show `value` will not be filled when it is not required.